src/core/src/service_interfaces/TelemetryWriter.py

```diff
@@ -14,131 +14,123 @@
+
+    def write_event_using_temp_file(self, folder_path, data, mode='w'):
+        """ Writes to a temp file in a single operation and then moves/overrides the original file with the temp """
+        file_path = self.__get_event_file_path(folder_path)
+        prev_events = []
         try:
-            payload = {'activity_id': str(self.activity_id), 'category': str(category), 'ver': "[%runbook_sub_ver%]", 'message': message}
-            payload = json.dumps(payload)[0:4095]
-            for transport in self.data_transports:
-                transport.write(payload)
-            return ""  # for consistency
+            if os.path.exists(file_path):
+                # if file_size exceeds max limit, sleep for 1 second, so the event can be written to a new file
+                file_size = self.get_file_size(file_path)
+                if file_size >= Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_BYTES:
+                    time.sleep(1)
+                    file_path = self.__get_event_file_path(folder_path)
+                else:
+                    prev_events = self.__fetch_events_from_previous_file(file_path)
+
+            prev_events.append(data)
+            with tempfile.NamedTemporaryFile(mode, dir=os.path.dirname(file_path), delete=False) as tf:
+                json.dump(prev_events, tf, default=data.__str__())
+                tempname = tf.name
+            shutil.move(tempname, file_path)
         except Exception as error:
-            return repr(error)  # if the caller cares
-
-    def close_transports(self):
-        """Close data transports"""
-        self.send_runbook_state_info('Closing telemetry channel(s).')
-        for transport in self.data_transports:
-            transport.close()
-    # endregion
-
-    # region Machine config retrieval methods
-    def get_machine_processor(self):
-        """Retrieve machine processor info"""
-        cmd = "cat /proc/cpuinfo | grep name"
-        code, out = self.env_layer.run_command_output(cmd, False, False)
-
-        if out == "" or "not recognized as an internal or external command" in out:
-            return "No information found"
-        # Example output:
-        # model name	: Intel(R) Core(TM) i7-6700 CPU @ 3.40GHz
-        lines = out.split("\n")
-        return lines[0].split(":")[1].lstrip()
-
-    def get_disk_type(self):
-        """ Retrieve disk info """
-        cmd = "cat /sys/block/sda/queue/rotational"
-        code, out = self.env_layer.run_command_output(cmd, False, False)
-        if "1" in out:
-            return "Hard drive"
-        elif "0" in out:
-            return "SSD"
-        else:
-            return "Unknown"
-    # end region
+            raise Exception("Unable to write to {0}. Error: {1}.".format(str(file_path), repr(error)))
+
+    def set_operation_id(self, operation_id):
+        self.operation_id = operation_id
```

Comment:
Why is this delayed set in Core? We shouldn't be writing out difficult to attribute events..